pkg/routers/router.go

```diff
@@ -109,3 +111,42 @@ func (r *LangRouter) Chat(ctx context.Context, request *schemas.UnifiedChatReque
 
 	return nil, ErrNoModelAvailable
 }
+
+// getPayload returns the payload based on the mathcing model from the given chat requests.
+//
+// It takes a slice of UnifiedChatRequest and a model string as parameters.
+// Returns a pointer to UnifiedChatRequest.
+func getPayload(request []schemas.UnifiedChatRequest, model string) *schemas.UnifiedChatRequest {
```

Comment:
Could we please pass structs as pointers to make sure they are not copied?

Reply:
Yep